### ISSUE TITLE
Colorer во Viewer 

### ISSUE THREAD

**Issue by zerochka** [elfmz/far2l#1234]
Было бы здорово если бы можно было бы как то, в конфиге или еще каким то хитрым способом включать колорер при просмотре файлов. 

Это бывает очень удобно при просмотре чужого кода. Viewer в отличии от редактора умеет умещать весь текст на экран, а если б он его еще и подсвечивал - было бы супер круто.

**Comment by akruphi** [comment]
Текущая логика здесь в том, что для просмотра файлов (например огромных логов) важна быстрота, а не замедление и парсенье. Причём просмотр far умеет обновляться при обновлении файла, при этом не блокируя файл, что особо ценно.

Но опционально по клавише (свободной в Viewer кажется осталась только **F9**) может и имеет смысл включать/выключать colorer, но никогда не по умолчанию.

**Comment by sviridov-alexey** [comment]
Текущая логика по моему у каждого своя, кто-то чаще смотрит огромные логи, а кто-то файлы исходников, не? Плюс Colorer же навешивается насколько я помню по расширению, почему *.cpp или *.java _НИКОГДА_ по умолчанию не должен  открываться с Colorer? Ну и на последок, перед открытием не оценить размер файла и показать предупреждение "Размер файла больше N, с в ключенной подсветкой это может долго, Продолжить без подсветки Да/Нет?"

**Comment by fred01** [comment]
Текущая логика по моему у каждого своя, кто-то чаще смотрит огромные логи, а кто-то файлы исходников, не? Плюс Colorer же навешивается насколько я помню по расширению, почему *.cpp или *.java НИКОГДА по умолчанию не должен открываться с Colorer? Ну и на последок, перед открытием не оценить размер файла и показать предупреждение "Размер файла больше N, с в ключенной подсветкой это может долго, Продолжить без подсветки Да/Нет?"

**Comment by elfmz** [comment]
ну в теории можно сделать `apt install source-highlight` и создать файл `.config/far2l/view.sh` с правами `chmod +x` и таким содержимым:
```
if [[ "$FILE" == *": "*" source"*" text"* ]] \
	|| [[ "$FILE" == *": "*" shell"*" text"* ]] \
	|| [[ "$FILE" == *": "*" script"*" text"* ]]; then

	source-highlight  --out-format=esc -i "$1" -o "$2"
	exit 0
fi
```
после чего во вьюере можно переключаться по F5 в режим с расцветкой, правда не колорером а этим самым source-highlight, но в принципе тоже симпатично

**Comment by elfmz** [comment]
хотя надо заметить wrap в таком режиме подглючивает неслабо..

**Comment by elfmz** [comment]
О, есть лучшее решение, Options/Viewer Settings.Viewer command: `source-highlight --out-format=esc -o STDOUT -i ./plugapi.cpp  | less -R`
После чего просматривать файлы по Alt+F3

**Comment by fred01** [comment]
Да, воркараунды есть. Да что далеко ходить, можно [bat](https://github.com/sharkdp/bat) поставить и его навешать хоть на Alt-F3, а можно в конце концов и в главное меню добавить. Но ишью по моему о другом, что бы без приседаний смотреть файлы по обычному F3 c уже идущим в поставке Colorer который имеет тесную интеграцию, настройки и т.д. По моему понятный use case. Но я думаю тут и без того фича-реквестов дофига, вряд ли это кто будет делать.

**Comment by elfmz** [comment]
У колорера в исходниках я видел режим сборки вьювера с запуском из командлайна, но это действительно все равно допиливать надо будет, так как надо переделать расцветку на escape sequences

**Comment by elfmz** [comment]
а впиливать интеграцию с колрером в стандартный вьювер путем расширения плагинового апи  это действительно задача не на ближайшее время

**Comment by fred01** [comment]
Все так, в конце концов кому надо - клавиатура в руки, это опенсорс, так это работает :)

**Comment by Vitalley** [comment]
Возможности в FAR не заложили вообще такие что бы colorer мог работать при просмотре?

**Comment by exkrexpexfex** [comment]
@zerochka  теперь включается в настройках colorer
